gandi/cli/commands/ip.py

```diff
@@ -123,32 +129,44 @@ def update(gandi, ip, reverse, background):
               help='This is a dangerous option that will cause CLI to continue'
                    ' without prompting. (default=False).')
 @pass_gandi
-def attach(gandi, ip, vm, background, force):
-    """Attach an ip to a vm.
+def attach(gandi, resources, vm, background, force):
+    """Attach one or more IPs to a vm.
 
-    ip can be an ip id or ip
+    ip can be an ip id or ip.
     vm can be a vm id or name.
     """
-    try:
-        ip_ = gandi.ip.info(ip)
-        vm_ = gandi.iaas.info(vm)
-    except UsageError:
-        gandi.error("Can't find this ip %s" % ip)
-
-    iface = gandi.iface.info(ip_['iface_id'])
-    if iface.get('vm_id'):
-        if vm_ and iface['vm_id'] == vm_.get('id'):
-            gandi.echo('This ip is already attached to this vm.')
-            return
 
-        if not force:
-            proceed = click.confirm('Are you sure you want to detach'
-                                    ' %s from vm %s' %
-                                    (ip_['ip'], iface['vm_id']))
-            if not proceed:
-                return
+    resources = sorted(tuple(set(resources)))
+    ret = []
+
+    for num, item in enumerate(resources):
+        try:
+            ip_ = gandi.ip.info(item)
+            vm_ = gandi.iaas.info(vm)
+        except UsageError:
+            gandi.error("Can't find this ip %s" % ip)
+            break
+
+        iface = gandi.iface.info(ip_['iface_id'])
+        if iface.get('vm_id'):
+            if vm_ and iface['vm_id'] == vm_.get('id'):
+                gandi.echo('IP %s is already attached to vm %s.' %
+                           (ip_['ip'], iface['vm_id']))
+                continue
+
+            if not force:
+                proceed = click.confirm('Are you sure you want to detach'
+                                        ' %s from vm %s?' %
+                                        (ip_['ip'], iface['vm_id']))
+                if not proceed:
+                    continue
+
+        gandi.echo('Attaching IP %s to VM %s...' %
+                    (ip_['ip'], iface['vm_id']))
+
+        ret.append(gandi.ip.attach(item, vm, background, force))
```

Comment:
Same here for the call to `gandi.ip.attach`, it will be hard to follow as it will made a progress bar per ip attach operations, You should update the module `IP.attach` method so it can handle multiple operations in input, as done in the vm module, so you can have one progress bar for all operations.